app/infrastructure/external/email/envelope_encryption.py

```
@@ -0,0 +1,156 @@
+            data_str = json.dumps(data) if isinstance(data, dict) else str(data)
+            dek = self._generate_dek()
+            key_id = self._generate_key_id()
+            fernet_dek = Fernet(dek)
+            encrypted_data = fernet_dek.encrypt(data_str.encode())
+            encrypted_dek = self._encrypt_dek(dek)
+            enc_dek_b64 = base64.urlsafe_b64encode(encrypted_dek).decode()
+            enc_data_b64 = base64.urlsafe_b64encode(encrypted_data).decode()
+            envelope = f"{key_id}:{enc_dek_b64}:{enc_data_b64}"
+            signature = self._sign_payload(envelope)
+            logger.debug("Encrypted data with key_id: %s", key_id)
+            return f"{envelope}:{signature}"
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)